### PR TITLE
Make docker-entrypoint take CRT and PEM 

### PR DESCRIPTION
**Pull request by skaymen**

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -2,7 +2,10 @@
 set -x
 
 keystorePassword=`cat $KEYSTORE_PASSWORD_FILE`
-keytool -v -importkeystore -srckeystore $WATER_AUTH_KEYS_FILE -srcstoretype PKCS12 -srcstorepass $keystorePassword -destkeystore $keystoreLocation -deststoretype JKS -deststorepass $keystorePassword -noprompt
+
+openssl pkcs12 -export -in $WATER_AUTH_CERT -inkey $WATER_AUTH_KEY -name water_auth_pkcs12 -out wa_pkcs12.p12
+keytool -v -importkeystore -deststorepass $keystorePassword -destkeystore $keystoreLocation -srckeystore wa_pkcs12.p12 -srcstoretype PKCS12
+keytool -v -import -alias bundle -trustcacerts -file wa_pkcs12.p12 -keystore $keystoreLocation
```
Comment (zmoore-usgs) [inline docker-entrypoint.sh]: I'm not sure if this third line is needed? I haven't done a lot of digging so I could certainly be missing something but I think the first line creates the PKCS12 keystore from the provided cert and key and then the second line converts the PKCS12 keystore into a JKS keystore which I think should be all that's necessary?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -2,7 +2,10 @@
 set -x
 
 keystorePassword=`cat $KEYSTORE_PASSWORD_FILE`
-keytool -v -importkeystore -srckeystore $WATER_AUTH_KEYS_FILE -srcstoretype PKCS12 -srcstorepass $keystorePassword -destkeystore $keystoreLocation -deststoretype JKS -deststorepass $keystorePassword -noprompt
+
+openssl pkcs12 -export -in $WATER_AUTH_CERT -inkey $WATER_AUTH_KEY -name water_auth_pkcs12 -out wa_pkcs12.p12
```
Comment (zmoore-usgs) [inline docker-entrypoint.sh]: This looks like a good method to use in general. For this application in particular there are 3 key pairs that need to be imported:

1. An Oauth2 Token Signing Key
2. A SAML Request Signing Key
3. A Tomcat SSL Key

These keys also need to have specific aliases within the keystore:

1. This alias should be the same as the value defined in the `keystoreDefaultKey` environment variable
2. This alias should be the same as the value defined in the `keystoreTokenSigningKey` environment variable
3. This alias should be `tomcat`

So you'll probably need 2 additional openssl import lines here to import those keys into the PKCS12 store before it is converted and you'll want to make sure that the aliases (which I think is what the `-name` flag is defining here? Or is that something else?) for all 3 match up as outlined above.

Comment (zmoore-usgs) [reply docker-entrypoint.sh]: Looking at the ticket I think the HTTPS key pair that @ayan-usgs is mentioning would be equivalent to the tomcat SSL key that I mentioned above, but might want to get that clarified to be sure that there aren't actually 4 key pairs that need to be imported.

**Comment by skaymen** [comment]
@zmoore-usgs made the suggested changes-- does that look right?

**Comment by ayan-usgs** [comment]
@zmoore-usgs and @skaymen -- I do happen to have some thoughts about getting one's hands on an existing PKCS12 store. One would have to connect to an instance that's running a water auth server container and copy the PKCS12 to the host machine. Once on the host machine, it can be transferred off using one's tool of choice.

**Pull request merged**